src/WebJobs.Extensions.ApiHub/Config/ApiHubConfiguration.cs

```diff
@@ -83,10 +88,18 @@ private IFolderItem GetFileSource(string key)
         /// Add path to the configuration
         /// </summary>
         /// <param name="key">App settings key name that have the connections string</param>
-        /// <param name="connectionString">Connection string to access SAAS via ApiHub. <seealso cref="ApiHubJobHostConfigurationExtensions.GetApiHubProviderConnectionStringAsync"/></param>
-        public void AddKeyPath(string key, string connectionString)
+        /// <param name="connectionString">Connection string to access SAAS via ApiHub. <seealso cref="ApiHubJobHostConfigurationExtensions.GetApiHubProviderConnectionStringAsync" /></param>
+        /// <param name="writer">The writer.</param>
+        public void AddKeyPath(string key, string connectionString, TraceWriter writer = null)
         {
-            var root = ItemFactory.Parse(connectionString);
+            ApiHubLogger logger = null;
+
+            if (writer != null)
+            {
+                logger = new ApiHubLogger(writer);
```

Comment:
Rather than newing up a logger each time this helper is called, can't you instead create the logger once outside so it doesn't have to be passed in by the caller?